qiime2/sdk/action.py

```diff
@@ -108,6 +108,22 @@ def _unmap_arg(arg, inputs):
     return unmapped
 
 
+@join_app
+def _deferred_alias(provenance, name, output, ctx, inputs=[]):
+    output = output.result()
+    return qiime2.sdk.util.create_future(
+        _alias(provenance, name, output, ctx))
```

Comment:
I would push the result of _alias up out of the expression. It blocks (which it should), but it looks at first like an expression that is minting a closure, (i.e. `map(make_apply(a, b,c), ...)`) rather than evaluating in place.

```python
output = output.result()
alias = _alias(provenance, name, output, ctx)
reutrn qiime2.sdk.util.create_future(alias)
```